File: ruby/lib/mable_bank/application/recording_logger.rb
# frozen_string_literal: true

module MableBank
  module Application
    class RecordingLogger
      include LoggerPort

      attr_reader :entries, :wire_entries

      def initialize
        @entries = []
        @wire_entries = []
      end

      def log_transfer_processed(telemetry)
        @entries << telemetry
        @wire_entries << telemetry.to_log_entry
      end
    end
  end
end

File: dotnet/src/MableBank.Application/RecordingLogger.cs
using MableBank.Application.Telemetry;

namespace MableBank.Application;

public sealed class RecordingLogger : ILogger
{
    public IList<TransferProcessedTelemetry> Entries { get; } = new List<TransferProcessedTelemetry>();

    public IList<TransferProcessedLogEntry> WireEntries { get; } = new List<TransferProcessedLogEntry>();

    public void LogTransferProcessed(TransferProcessedTelemetry telemetry)
    {
        ArgumentNullException.ThrowIfNull(telemetry);
        Entries.Add(telemetry);
        WireEntries.Add(telemetry.ToLogEntry());
    }
}
